In [1]:
%pip install git+https://github.com/awslabs/fast-differential-privacy

  Cloning https://github.com/awslabs/fast-differential-privacy to /tmp/pip-req-build-hu1asloz
  Running command git clone --filter=blob:none --quiet https://github.com/awslabs/fast-differential-privacy /tmp/pip-req-build-hu1asloz
  Resolved https://github.com/awslabs/fast-differential-privacy to commit 1202baa441c38c338415bc24e4cf12514c46bac0
  Preparing metadata (setup.py) ... done
  Created wheel for fastDP: filename=fastDP-1.1.0-py3-none-any.whl size=101664 sha256=90c60394fb73ba8a0b99e37e99871a9eaaf32780dd014b629e8da3872e22ca79
  Stored in directory: /tmp/pip-ephem-wheel-cache-3nb77txc/wheels/b3/2b/b5/62668a08336b29e28ca790358738474aa8c2c68359b494f454
Successfully built fastDP


In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
%cd /content/drive/Othercomputers/My Mac/NormaLcifar10/

/content/drive/Othercomputers/My Mac/NormaLcifar10


In [ ]:
!python linearprobing.py --name fttrial --seed 1 --epoch 50 --epsilon 1 --b 810 --lr 1 --g 0.8

Files already downloaded and verified
Files already downloaded and verified
Downloading: "https://download.pytorch.org/models/resnet18-f37072fd.pth" to /root/.cache/torch/hub/checkpoints/resnet18-f37072fd.pth
100% 44.7M/44.7M [00:00<00:00, 149MB/s]
Total parameters 11689512. Trainable parameters 513000.
Number of trainable components:  2 ; Number of trainable layers:  1
>>>>>>>>>>>>>>>>> Applying  automatic  per-sample gradient clipping.
>>>>>>>>>>>>>>>>> Block heads for per-sample gradient clipping are defined as: ['fc']
noise multiplier 3.7438720703125004
/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py:1359: UserWarning: Using a non-full backward hook when the forward contains multiple autograd Nodes is deprecated and will be removed in future versions. This hook will be missing some grad_input. Please use register_full_backward_hook to get the documented behavior.
  warnings.warn("Using a non-full backward hook when the forward contains multiple autograd Nodes "


- !python linearprobing.py --name lptrial --seed 1 --epoch 50 --epsilon 1.5 --b 810 --lr 1 --g 0.9

  61 %,70 %,72 %,74 %,74 %,75 %,75 %,76 %,76 %,76 %,76 %,76 %,76 %,76 %,77 %,77 %

- !python linearprobing.py --name lptrial --seed 1 --epoch 50 --epsilon 1 --b 810 --lr 1 --g 0.9

  noise multiplier 3.7438720703125004

  59 %,69 %,71 %,73 %,74 %,74 %,75 %,75 %,75 %,75 %,76 %,76 %,76 %,76 %,76 %,76 %,76 %,76 %,77 %,77 %,77 %